### PR TITLE
Task/ckan utils migration

### PR DESCRIPTION
**Pull request by b-rvc**
Migrated Utils for CKAN

### CODE REVIEW

File: iotqatools/ckan_utils.py [hunk 1]
```diff
@@ -0,0 +1,184 @@
+
+
+import json
+import requests
+import ast
+from iotqatools.iot_logger import get_logger
+
+from behave import *
```
Comment (manucarrace) [inline iotqatools/ckan_utils.py]: Import not needed

File: iotqatools/ckan_utils.py [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+from nose.tools import assert_true
+from iotqatools.iot_tools import PqaTools
+
+__logger__ = get_logger("CKAN_utils", 'DEBUG', True)
+
+class CkanUtils(object):
```
Comment (manucarrace) [inline iotqatools/ckan_utils.py]: Expected two blanck lines before. Please use PEP8

File: iotqatools/ckan_utils.py [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+
+    @staticmethod
+    def get_last_value_from_resource(context, endpoint, apikey, resourceid):
```
Comment (manucarrace) [inline iotqatools/ckan_utils.py]: parameters endpoint, apikey and resourceid not used. Are really needed?

File: iotqatools/ckan_utils.py [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+            result = json.loads(context.config['components']['CKAN']['response'].content)['result']
+
+        except requests.exceptions.RequestException as e:
+            __logger__.info("Network error: " + e)
+            assert_true(False, msg="[NETWORK ERROR]")
+
+        return result
+    
+    @staticmethod
+    def ckanCreateOrg(context, comp, service, secure=False):
```
Comment (manucarrace) [inline iotqatools/ckan_utils.py]: Function names should be lowercase. Use "_"

File: iotqatools/ckan_utils.py [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+            context.config[comp]['response'] = 'ERROR'
+            __logger__.info('[ERROR] '+ e)
+            assert_true(False, msg='[NETWORK ERROR]')
+
+            # Show info in logs
+            # PqaTools.log_result(url=url, headers=headers, data=data, comp=comp)
+            # TODO: show log info
+
+    @staticmethod
+    def ckanCreatePkg(context, comp, service, secure=False):
```
Comment (manucarrace) [inline iotqatools/ckan_utils.py]: Use lowercase in functions name

File: iotqatools/ckan_utils.py [hunk 6]
```diff
@@ -0,0 +1,184 @@
...
+            context.config['components'][comp]['response'] = 'ERROR'
+            __logger__.info("Network error: " + e)
+            assert_true(False, msg='[NETWORK ERROR]')
+
+            # Show info in logs
+            # PqaTools.log_result(url=url, headers=headers, data=data,comp=comp)
+            # TODO: Show info in logs
+
+    @staticmethod
+    def ckanCreateRsc(context, comp, service, secure=False):
```
Comment (manucarrace) [inline iotqatools/ckan_utils.py]: Use lowercase

File: iotqatools/ckan_utils.py [hunk 7]
```diff
@@ -0,0 +1,184 @@
...
+            context.config['components'][comp]['response'] = 'ERROR'
+            __logger__.info("Network error: " + e)
+            assert_true(False, msg='[NETWORK ERROR]')
+
+            # Show info in logs
+            # PqaTools.log_result(url=url, headers=headers, data=data,comp=comp)
+            # TODO: Show info in logs
+
+    @staticmethod
+    def ckanGetRsc(context, comp, service, secure=False):
```
Comment (manucarrace) [inline iotqatools/ckan_utils.py]: Use lowercase

File: iotqatools/ckan_utils.py [hunk 8]
```diff
@@ -0,0 +1,184 @@
...
+        # Show info in logs
+        __logger__.info(" > RESPONSE: {}".format(context.config['components'][comp]['response']))
+        result = json.loads(context.config['components'][comp]['response'].content)['result']
+
+        # print result["resources"][0]["id"]
+        if context.config['components'][comp]['response'].status_code == 200:
+            PqaTools.remember('resource-id', result['resources'][0]['id'])
+
+    @staticmethod
+    def ckanGetPkg(context, comp, service, secure=False):
```
Comment (manucarrace) [inline iotqatools/ckan_utils.py]: Use lowercase

File: iotqatools/ckan_utils.py [hunk 9]
```diff
@@ -0,0 +1,184 @@
...
+            __logger__.info("Network error: " + e)
+            assert_true(False, msg='[NETWORK ERROR]')
+
+            # Show info in logs
+            # if world.vm:
+            # PqaTools.log_result(url=url, headers=headers, data=data,comp=comp)
+            # TODO: Show info in logs
+
+    @staticmethod
+    def ckanCreateDs(context, comp, service, secure=False):
```
Comment (manucarrace) [inline iotqatools/ckan_utils.py]: Use lowercase

File: iotqatools/ckan_utils.py [hunk 10]
```diff
@@ -0,0 +1,184 @@
...
+            context.config['components'][comp]['response'] = 'ERROR'
+            __logger__.info("Network error: " + e)
+            assert_true(False, msg='[NETWORK ERROR]')
+
+            # Show info in logs
+            # PqaTools.log_result(url=url, headers=headers, data=data,comp=comp)
+            # TODO: Show info in logs
+
+    @staticmethod
+    def ckanCreateDs2(context, comp, service, secure=False):
```
Comment (manucarrace) [inline iotqatools/ckan_utils.py]: use lowercase

File: iotqatools/ckan_utils.py [hunk 11]
```diff
@@ -0,0 +1,184 @@
...
+                                                                       fields, verify_ssl=verify_ssl)
+        except requests.exceptions.RequestException, e:
+            context.config['components'][comp]['response'] = 'ERROR'
+            __logger__.info("Network error: " + e)
+            assert_true(False, msg='[NETWORK ERROR]')
+
+
+
+    @staticmethod
+    def ckanDeleteOrg(context, comp, service, secure=False):
```
Comment (manucarrace) [inline iotqatools/ckan_utils.py]: Use lowercase

File: iotqatools/ckan_utils.py [hunk 12]
```diff
@@ -0,0 +1,184 @@
...
+        try:
+            context.config['components'][comp]['response'] = context.o[comp].create_datastore(context.remember['resource'], context.remember['package'],
+                                                                       fields, verify_ssl=verify_ssl)
+        except requests.exceptions.RequestException, e:
+            context.config['components'][comp]['response'] = 'ERROR'
+            __logger__.info("Network error: " + e)
+            assert_true(False, msg='[NETWORK ERROR]')
+
+
+
```
Comment (manucarrace) [inline iotqatools/ckan_utils.py:374]: Two many blanck lines.

Comment (b-rvc) [inline iotqatools/ckan_utils.py:374]: Fixed

File: iotqatools/ckan_utils.py (hunk 11 above)
Comment (b-rvc) [inline iotqatools/ckan_utils.py]: PEP8 used for all method naming.

File: iotqatools/ckan_utils.py (hunk 1 above)
Comment (b-rvc) [inline iotqatools/ckan_utils.py]: Needed for importing the context although pycharm says it's not used

**Pull request closed without merging**